ctsit/report_handler
# Feature/migrate to poetry

**What does this change do?** 

This change migrates the project to use `poetry` for dependency management.

**Why was this change made?** 

To align dependency management with other projects.

## Verification
1. Verify that all automated tests pass
2. Update the `report_handler` dependency in `PyCustodian`
  - <img width="967" alt="image" src="https://github.com/ctsit/report_handler/assets/75623654/1da11229-4830-4050-a54e-efb7294a992d">
3. Update dependencies with `poetry install`
4. Run the sample script and verify that a `xlsx` report is generated that contains entries.

## Affirmations

All of these should have a check by them. Any exception requires an explanation.
